### PR TITLE
WT-6916 Create stub subpages for architecture guide.

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Comment by ddanderson** [comment]
@keitharnoldsmith, I know this is late on a Friday - there's no rush, it can be reviewed next week.

**Comment by ddanderson** [comment]
If you want to "see" the pages, you'll need to pull the branch and run `s_docs` in `dist/`.

File: src/docs/arch-metadata.dox [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*! @arch_page arch-metadata Metadata
+
+Metadata in WiredTiger is stored as a table in the \c "WiredTiger.wt" file.
+The table's key is a \c uri string, and the value is its complete
+configuration string.  The configuration itself is a list of key/value
+pairs in string form.  The configuration's keys are dependent on the
+type of \c uri.  Thus, a metadata entry with a \c uri key beginning with
+\c "table:" will a configuration string
```
Comment (keitharnoldsmith) [inline src/docs/arch-metadata.dox:8]: "will a configuration string" --> "will be a configuration string" ?

File: src/docs/arch-index.dox [hunk 2]
```diff
@@ -121,15 +123,75 @@ wt_file -[hidden]right-> log_file
...
+
+Metadata is stored as <code>uri, config</code> K/V pairs in a designated table.
+
+@subpage arch-python
+
+WiredTiger has a Python API that is useful for scripting and experimentation.
+
+@subpage arch-row
+
+Column Stores are Btrees that have a variable size key and data.
```
Comment (keitharnoldsmith) [inline src/docs/arch-index.dox:184]: "Column Stores" --> "Row Stores"

File: src/docs/arch-eviction.dox [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/*! @arch_page arch-eviction Eviction
+
+Eviction represents the process of writing dirty data to disk.
```
Comment (keitharnoldsmith) [inline src/docs/arch-eviction.dox:3]: Doesn't eviction cover include reclaiming space from clean pages, as well?

File: src/docs/arch-index.dox [hunk 4]
```diff
@@ -121,15 +123,75 @@ wt_file -[hidden]right-> log_file
...
+The format of the data file is given by structures in \c block.h .
+
 @subpage arch-dhandle
 
 An internal structure called a Data Handle (dhandle) is used to represent and
 access Btrees and other data sources in WiredTiger.
 
+@subpage arch-eviction
+
+Eviction represents the process of writing dirty data to disk.
```
Comment (keitharnoldsmith) [inline src/docs/arch-index.dox:154]: See note above.  Maybe, "Eviction represents the process or removing old data from the cache, writing it to disk if it is dirty"?

**Review by keitharnoldsmith — APPROVED** [review]
LGTM!

I've left a couple of comments on typos and other nits.  You'll probably want to fix them, but since these are stub files you could also leave them for whoever fills in the stub!  😄

File: src/docs/arch-eviction.dox (hunk 3 above)
Comment (ddanderson) [reply src/docs/arch-eviction.dox:3]: Yes, added that.

File: src/docs/arch-index.dox (hunk 4 above)
Comment (ddanderson) [reply src/docs/arch-index.dox:154]: I reworked this slightly, starting with your words.

**Pull request merged**